cluster-autoscaler/core/scale_down_test.go

```diff
@@ -514,6 +515,34 @@ func TestDeleteNode(t *testing.T) {
 					fakeNode = obj.DeepCopy()
 					return true, obj, nil
 				})
+			fakeClient.Fake.AddReactor("patch", "nodes",
+				func(action core.Action) (bool, runtime.Object, error) {
+					patch := action.(core.PatchAction)
+					patchMap := make(map[string]interface{})
+					if err := json.Unmarshal(patch.GetPatch(), &patchMap); err != nil {
+						return false, nil, err
+					}
+					// Ugly but works for retrieving taints from patch.
```

Comment:
Isn't there some utility (maybe in pkg/strategicpatch module) to apply a patch. It would be much more cleaner.

Reply:
Not that I could find any, but if you do, I'm happy to update the PR. Note this is only in test code (I wouldn't use this approach in production path).

Reply:
`StrategicMergePatchUsingLookupPatchMeta`?

Reply:
Also I am not a big fan of having different code quality standards for tests code and prod code. 
We spend a lot of time changing test code and we do not want to put a lot of tech-debt there as it basically slows us down.